lz4.go

```diff
@@ -291,3 +299,212 @@ func (r *reader) readFromPending(dst []byte) (int, error) {
 	}
 	return copied, nil
 }
+
+// CompressReader reads input and creates an io.ReadCloser for reading
+// compressed output
+type CompressReader struct {
+	underlyingReader       io.Reader
+	compressionBuffer      [2]unsafe.Pointer
+	outputBuffer           *bytes.Reader
+	lz4Stream              *C.LZ4_stream_t
+	inpBufIndex            int
+	totalCompressedWritten int
+	compressedBuffer       unsafe.Pointer
+}
+
+// NewCompressReader creates a new io.ReadCloser.  Reads from the returned ReadCloser
+// read and compress data from r.  It is the caller's responsibility to call
+// Close on the ReadCloser when done.  If this is not done, underlying objects
+// in the lz4 library will not be freed.
+func NewCompressReader(r io.Reader) *CompressReader {
+	return &CompressReader{
+		compressionBuffer: [2]unsafe.Pointer{
+			C.malloc(hugeStreamingBlockSize),
+			C.malloc(hugeStreamingBlockSize),
+		},
+		lz4Stream:        C.LZ4_createStream(),
+		underlyingReader: r,
+		outputBuffer:     bytes.NewReader(nil),
+		compressedBuffer: C.malloc(boundedHugeStreamingBlockSize + blockHeaderSize),
+	}
+}
+
```

Comment:
We can get rid of this now, right?
https://github.com/DataDog/dd-go/blob/95490eec042ffec45479b4450ee02420390cf363/apps/spidly/snapshot/file.go#L165-L175

Reply:
yes, we can! @flapjack103 has a PR that removes it :smile: